### PR TITLE
94 create lessons page

### PR DESCRIPTION
**Pull request by ayomanetsan**
Update lesson creation component according to the new design. In order for new fields to be accepted correctly, I updated the Lesson Service as well as Lesson entity fields.

image

### CODE REVIEW

**Comment by Anatolii-Ivanov** [comment]
Please use our buttons)

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
             lessonsFromContext = lessonsFromContext.Where(m => filtersRequest.LanguageLevels.Contains(m.LanguageLevel));
         }
 
-        var subscribersCountDict = await lessonsFromContext.Select(t => new {t.Id, SbCount = t.Subscribers.Count }).ToDictionaryAsync(t => t.Id);
+        var subscribersCountDict = await lessonsFromContext.Select(t => new { Id = t.Id, SbCount = t.LimitOfUsers ?? 0 }).ToDictionaryAsync(t => t.Id);
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:49]: But why? We show amount of registered users

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html [hunk 2]
```diff
@@ -1,56 +1,121 @@
...
+            <mat-error *ngIf="questions?.invalid && questions?.touched">
+                At least one question is required
             </mat-error>
-        </mat-form-field>
+        </div>
 
-        <button mat-raised-button [disabled]="myForm.invalid" color="primary" (click)="createLesson()">Create a lesson</button>
+        <!-- TODO: Update create button when design is finished -->
 
+        <button mat-flat-button color="primary" (click)="createLesson()">SAMPLE CREATE BUTTON</button>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html:119]: Change to create lesson
If we won't have it in design, we won't need to fix it at least

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts [hunk 3]
```diff
@@ -50,23 +65,67 @@ export class LessonsCreateComponent implements OnInit {
...
+    expandLevelDropdown() {
+        this.levelDropdownVisible = !this.levelDropdownVisible;
+    }
+
+    updateTags(evendData: string[]) {
+        this.tagsList = evendData;
+    }
+
+    getYoutubeVideoId(link: string): string {
+        const regex = /(?:youtube(?:-nocookie)?\.com\/(?:[^/\n\s]+\/\S+\/|(?:v|vi|user)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts:93]: Move regex to file with regex

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts [hunk 4]
```diff
@@ -50,23 +65,67 @@ export class LessonsCreateComponent implements OnInit {
...
     createLesson() {
+        this.submitted = true;
+
+        if (this.myForm.invalid || !this.time || !this.level || !this.tagsList.length) {
+            return;
+        }
+
+        const timeHours = parseInt(this.time.split('.')[0], 10);
+        const timeMinutes = parseInt(this.time.split('.')[1], 10);
+        const startAt = new Date(new Date(this.date?.value).setHours(timeHours, timeMinutes));
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts:108]: Use Moment please

File: frontend/src/app/shared/utils/lesson.utils.ts [hunk 5]
```diff
@@ -22,91 +20,39 @@ export default class Utils {
...
+        '15.30',
+        '16.00',
+        '16.30',
+        '17.00',
+        '17.30',
+        '18.00',
+        '18.30',
+        '19.00',
+        '19.30',
+        '20.00',
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/utils/lesson.utils.ts:48]: Can we let user select time? As we will need the whole range + I see no reason why we can't create lesson at 8.15

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by NadiaMaksymchuk — APPROVED** [review]

**Pull request merged**